tests/chebyshev_test.py

```diff
@@ -0,0 +1,812 @@
+      "dtype1": dtype1, "length1": length1, "trailing1": trailing1,
+      "dtype2": dtype2, "length2": length2, "trailing2": trailing2}
+      for dtype1 in all_dtypes
+      for dtype2 in all_dtypes
+      for length1 in [1, 5, 10]
+      for length2 in [1, 5, 10]
+      for trailing1 in [0, 1]
+      for trailing2 in [0, 1]))
+  def testChebmul(self, dtype1, dtype2, length1, length2, trailing1, trailing2):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      c1 = rng((length1,), dtype1)
+      c2 = rng((length2,), dtype2)
+      return (jnp.concatenate([c1, jnp.zeros(trailing1, c1.dtype)]),
+              jnp.concatenate([c2, jnp.zeros(trailing2, c2.dtype)]))
+
+    jnp_fn = partial(cs.chebmul, trim=False)
+
+    with jtu.strict_promotion_if_dtypes_match([dtype1, dtype2]):
+      self._CheckAgainstNumpy(ncs.chebmul, cs.chebmul, args_maker, check_dtypes=False,
+                              tol=1e-6)
+      self._CompileAndCheck(jnp_fn, args_maker)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": """_dtype1={}_trailing1={}
+    _dtype2={}_trailing2={}""".format(
+      jtu.format_shape_dtype_string((length1+trailing1,), dtype1),
+      trailing1,
+      jtu.format_shape_dtype_string((length2+trailing1,), dtype2),
+      trailing2),
+      "dtype1": dtype1, "length1": length1, "trailing1": trailing1,
+      "dtype2": dtype2, "length2": length2, "trailing2": trailing2}
+      for dtype1 in all_dtypes
+      for dtype2 in all_dtypes
+      for length1 in [1, 5, 10]
+      for length2 in [1, 5, 10]
+      for trailing1 in [0, 1]
+      for trailing2 in [0, 1]))
+  def testChebdiv(self, dtype1, dtype2, length1, length2, trailing1, trailing2):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      c1 = rng((length1,), dtype1)
+      c2 = rng((length2,), dtype2)
+
+      return (jnp.concatenate([c1, jnp.zeros(trailing1, c1.dtype)]),
+              jnp.concatenate([c2, jnp.zeros(trailing2, c2.dtype)]))
+
+    jnp_fn = partial(cs.chebdiv, trim=False)
+
+    with jtu.strict_promotion_if_dtypes_match([dtype1, dtype2]):
+      self._CheckAgainstNumpy(ncs.chebdiv, cs.chebdiv, args_maker, check_dtypes=False,
+                              tol=1e-6)
+      self._CompileAndCheck(jnp_fn, args_maker)
+
+  @parameterized.named_parameters(jtu.cases_from_list(
+    {"testcase_name": "_dtype={}_trailing={}_pow={}".format(
+      jtu.format_shape_dtype_string((length+trailing,), dtype),
+      trailing, pow),
+      "dtype": dtype, "length": length, "trailing": trailing, "pow": pow}
+      for dtype in all_dtypes
+      for length in [1, 5, 10]
+      for trailing in [0, 1]
+      for pow in [0,1,2,10]))
+  def testChebpow(self, dtype, length, trailing, pow):
+    rng = jtu.rand_default(self.rng())
+
+    def args_maker():
+      c = rng((length,), dtype)
+      return (jnp.concatenate([c, jnp.zeros(trailing, c.dtype)]), pow)
+
+    jnp_fn = lambda c, _p: partial(cs.chebpow, pow=pow, trim=False)(c)
```

Comment:
I'd use either `lambda` or `partial`, but not both together, as it is a bit confusing to parse.

Same in several places below.

Reply:
Also, it's strange here that the tested function ignores the second input. If it needs to be static, then we shouldn't generate it as part of `args_maker`, but rather use a `partial` for both the jax and numpy version of the function. You can see that pattern elsewhere in the test suite when testing static arguments.

Same comment applies to many of the test cases below.